src/components/SharedComponents/Map/Map.tsx

```diff
@@ -148,12 +148,36 @@ const Map = ( {
     }
   }
 
+  // In Android, we maintain a state for defaultInitialRegion and initialRegion
+  // that is updated on gestures like pan and zoom. This state is always null in iOS.
+  const [localRegion, setLocalRegion] = useState<Region|null>( Platform.OS === "android"
+    ? initialRegion || defaultInitialRegion
+    : null );
+
+  // In Android, onMapReady does not fire when we pass parameter region instead
+  // of parameter initialRegion. This state allows us to fire onMapReady and
+  // fire it only once. This state is always false in iOS.
+  const [onMapReadyHasFired, setOnMapReadyHasFired] = useState( false );
```

Comment:
Same here. This will be very confusing if for some reason we need to know if `onMapReady` has fired in iOS. The reader's assumption should be that it has if `onMapReadyHasFired` is true given the name, but it will be false. As above, I'd prefer this OS-specific functionality to be in a separate file, but failing that I think `onMapReadyHasFiredAndroid` or something would be more explicit.